packages/aws-cdk/lib/api/logs/logs-monitor.ts

```diff
@@ -0,0 +1,209 @@
+   */
+  private readonly envsLogGroupsAccessSettings = new Map<string, LogGroupsAccessSettings>();
+
+  private active = false;
+
+  constructor(startTime?: Date) {
+    this.startTime = startTime?.getTime() ?? Date.now();
+  }
+
+  /**
+   * resume reading/printing events
+   */
+  public activate(): void {
+    this.active = true;
+    this.scheduleNextTick(0);
+  }
+
+  /**
+   * deactivates the monitor so no new events are read
+   * use case for this is when we are in the middle of performing a deployment
+   * and don't want to interweave all the logs together with the CFN
+   * deployment logs
+   *
+   * Also resets the start time to be when the new deployment was triggered
+   * and clears the list of tracked log groups
+   */
+  public deactivate(): void {
+    this.active = false;
+    this.startTime = Date.now();
+    this.envsLogGroupsAccessSettings.clear();
+  }
+
+  /**
+   * Adds CloudWatch log groups to read log events from.
+   * Since we could be watching multiple stacks that deploy to
+   * multiple environments (account+region), we need to store a list of log groups
+   * per env along with the SDK object that has access to read from
+   * that environment.
+   */
+  public addLogGroups(env: cxapi.Environment, sdk: ISDK, logGroupNames: string[]): void {
+    const awsEnv = `${env.account}:${env.region}`;
+    const logGroupsStartTimes = logGroupNames.reduce((acc, groupName) => {
+      acc[groupName] = this.startTime;
+      return acc;
+    }, {} as { [logGroupName: string]: number });
+    this.envsLogGroupsAccessSettings.set(awsEnv, {
+      sdk,
+      logGroupsStartTimes: {
+        ...this.envsLogGroupsAccessSettings.get(awsEnv)?.logGroupsStartTimes,
+        ...logGroupsStartTimes,
+      },
+    });
+  }
+
+  private scheduleNextTick(sleep: number): void {
+    setTimeout(() => void(this.tick()), sleep);
+  }
+
+  private async tick(): Promise<void> {
+    if (!this.active) {
+      return;
+    }
+    try {
+      const events = flatten(await this.readNewEvents());
+      events.forEach(event => {
+        this.print(event);
+      });
+    } catch (e) {
+      error('Error occurred while monitoring logs: %s', e);
+    }
+    this.scheduleNextTick(SLEEP);
+  }
+
+  /**
+   * Reads all new log events from a set of CloudWatch Log Groups
+   * in parallel
+   */
+  private async readNewEvents(): Promise<Array<Array<CloudWatchLogEvent>>> {
+    const promises: Array<Promise<Array<CloudWatchLogEvent>>> = [];
+    for (const settings of this.envsLogGroupsAccessSettings.values()) {
+      for (const group of Object.keys(settings.logGroupsStartTimes)) {
+        promises.push(this.readEventsFromLogGroup(settings, group));
+      }
+    }
+    return Promise.all(promises);
+  }
+
+  /**
+   * Print out a cloudwatch event
+   */
+  private print(event: CloudWatchLogEvent): void {
+    print(util.format('[%s] %s %s',
+      colors.blue(event.logGroupName),
+      colors.yellow(event.timestamp.toLocaleTimeString()),
```

Comment:
What timezone are these timestamps in? Are they correct? Wouldn't we want more control over the formatting?

Reply:
The timestamp of the event is using UTC (expressed as the number of milliseconds after Jan 1, 1970 00:00:00 UTC) so this should just format those to the users locale. So for example someone in the US would get something like `12/11/2012, 7:00:00 PM`.

Reply:
Yes, everyone will see the time printed in their local timezone, which I think is what you want here.

Reply:
Pretty sure `toLocaleTimeString()` does not do timezone conversion, and I think native d/t formatting is not a great idea.

But it will do for now, I suppose.

Reply:
I definitely see the logs in PST, so it's either `toLocaleTimeString()`, or perhaps CloudWatch returns them in the correct locale...?

Reply:
I tested on machines with different timezones and I get whatever the local time is.